src/CLRunner.ts

```diff
@@ -335,6 +335,110 @@ export class CLRunner {
+            // For local stroate logDialogId = sessionId
+            const logDialogId = sessionId
+
+            // Append an extractor step to already existing log dialog
+            const logDialog: CLM.LogDialog | undefined = await ConversationLearner.logStorage.Get(appId, logDialogId)
+            if (!logDialog) {
+                throw new Error(`Log Dialog does not exist App:${appId} Id:${logDialogId}`)
+            }
+            const newRound: CLM.LogRound = {
+                extractorStep: { ...extractResponse, stepBeginDatetime, stepEndDatetime },
+                scorerSteps: []
+            }
+            logDialog.rounds.push(newRound)
+            await ConversationLearner.logStorage.Replace(appId, logDialog)
+        }
+        return extractResponse
+    }
+
+    private async SessionScore(appId: string, sessionId: string, scoreInput: CLM.ScoreInput): Promise<CLM.ScoreResponse> {
+        const stepBeginDatetime = new Date().toJSON()
+        const scoreResponse = await this.clClient.SessionScore(appId, sessionId, scoreInput)
+
+        // Add to dev's log storage account (if it exists)
+        if (ConversationLearner.logStorage) {
+            // For self-hosted storage logDialogId is sessionId
+            const logDialogId = sessionId
+            const predictedAction = scoreResponse.scoredActions[0] ? scoreResponse.scoredActions[0].actionId : ""
+
+            // Keep only needed data (drop payload, etc)
+            const scoredActions = scoreResponse.scoredActions.map(sa => {
+                return {
+                    score: sa.score,
+                    actionId: sa.actionId
+                }
+            })
+            const unscoredActions = scoreResponse.unscoredActions.map(sa => {
+                return {
+                    reason: sa.reason,
+                    actionId: sa.actionId
+                }
+            })
+            // Need to use recursive partial as scored and unscored have only partial data
+            const logScorerStep: Utils.RecursivePartial<CLM.LogScorerStep> = {
+                input: scoreInput,
+                predictedAction,
+                logicResult: undefined, // LARS what should this be
+                predictionDetails: { scoredActions, unscoredActions },
+                stepBeginDatetime,
+                stepEndDatetime: new Date().toJSON(),
+                metrics: scoreResponse.metrics
+            }
+
+            const logDialog: CLM.LogDialog | undefined = await ConversationLearner.logStorage.Get(appId, logDialogId)
+            if (!logDialog) {
+                throw new Error(`Log Dialog does not exist App:${appId} Log:${logDialogId}`)
+            }
+            const lastRound = logDialog.rounds[logDialog.rounds.length - 1]
+            if (!lastRound || !lastRound.extractorStep) {
+                throw new Error(`Log Dialogs has no Extractor Step Id:${logDialogId}`)
+            }
+            lastRound.scorerSteps.push(logScorerStep as any)
```

Comment:
Same as above. a helper function that gets the scorerResponse and adds it as scorerStep to logDialog

Reply:
As above...